### PR TITLE
Login form Pt. 1/x

### PR DESCRIPTION
**Pull request by DanielW1987**
Here is what I implemented for the login form:

- Using `react-hook-form`. Ideally we should use this everywhere we work with forms.
- Client-side validation with `react-hook-form`.
- Dedicated react hook for client <-> server communication (`useSignIn`) with error handling, loading state and in-memory persistence of auth context via context API
- After a refresh the Auth State is gone
- Works only on localhost for now
- First preparations for environment specific URLs are already done, see .env.development and .env.production
- Loading spinner is on the login button
- All todos are intentionally still open

There will be follow up PRs

### CODE REVIEW

File: src/Components/Auth/Login/LoginForm.module.scss [hunk 1]
```diff
@@ -10,4 +10,4 @@
     flex-direction: row;
     justify-content: space-between;
   }
-}
+}
```
Comment (hedgehog1833) [inline src/Components/Auth/Login/LoginForm.module.scss:13]:
```suggestion
}

```

File: src/Api/Axios.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Axios from 'axios'
+import { PathLike } from 'fs'
+import qs from 'qs'
+
+export default Axios.create({
```
Comment (hedgehog1833) [inline src/Api/Axios.ts:5]: Why have two axios configs?

**Review by hedgehog1833 — APPROVED** [review]
A lot in here but I guess it look good. Some questions but they can be answered later :)

File: src/Api/Axios.ts (hunk 2 above)
Comment (DanielW1987) [reply src/Api/Axios.ts:5]: It was easier to start with a new config. The configs will be combined into one later.

**Pull request merged**